code/modules/mob/living/silicon/robot/robot.dm

```diff
@@ -995,3 +997,89 @@
 	hat = new_hat
 	new_hat.forceMove(src)
 	update_icons()
+
+/mob/living/silicon/robot/MouseDrop_T(mob/living/M, mob/living/user)
+	. = ..()
+	if(!(M in buckled_mobs))
+		buckle_mob(M)
+
+/mob/living/silicon/robot/buckle_mob(mob/living/M, force = 0, check_loc = 1)
+	if(!riding_datum)
+		riding_datum = new /datum/riding/cyborg
+		riding_datum.ridden = src
+	if(buckled_mobs)
+		if(buckled_mobs.len >= max_buckled_mobs)
+			return
+		if(M in buckled_mobs)
+			return
+	if(stat)
+		return
+	if(incapacitated())
+		return
+	if(M.restrained())
+		return
+	if(iscyborg(M))
+		M.visible_message("<span class='warning'>[M] really can't seem to mount the [src]...</span>")
+		return
+	if(isbot(M))
+		M.visible_message("<span class='boldwarning'>No. Just... no.</span>")
+		return
+	if(module)
+		if(!module.allow_riding)
+			M.visible_message("<span class='boldwarning'>Unfortunately, [M] just can't seem to hold onto [src]!</span>")
+			return
+	if(iscarbon(M))
+		if(!equip_buckle_inhands(M))	//MAKE SURE THIS IS LAST!
+			M.visible_message("<span class='boldwarning'>[M] can't climb onto [src] because his hands are full!</span>")
+			return
+	. = ..(M, force, check_loc)
+	riding_datum.handle_vehicle_offsets()
+
+/mob/living/silicon/robot/unbuckle_mob(mob/user)
+	if(iscarbon(user))
+		unequip_buckle_inhands(user)
+	. = ..(user)
+	riding_datum.restore_position(user)
+
+/mob/living/silicon/robot/proc/unequip_buckle_inhands(mob/living/carbon/user)
+	for(var/obj/item/cyborgride_offhand/O in user.contents)
+		if(O.ridden != src)	//This shouldn't EVER happen.
+			continue
+		if(O.selfdeleting)
+			continue
+		else
+			qdel(O)
+	return TRUE
+
+/mob/living/silicon/robot/proc/equip_buckle_inhands(mob/living/carbon/user)
+	var/obj/item/cyborgride_offhand/inhand = new /obj/item/cyborgride_offhand(user)
+	inhand.rider = user
+	inhand.ridden = src
+	return user.put_in_hands(inhand, TRUE)
+
+/obj/item/cyborgride_offhand
```

Comment:
Just an idea; but what if this item's throw_at unbuckled the rider and then called throw_at on them with the same args